pytorch_lightning/trainer/trainer.py

```diff
@@ -422,6 +422,14 @@ def __init__(
         self.optimizer_connector.on_trainer_init()
 
         # init data flags
+        if prepare_data_per_node is None:
+            prepare_data_per_node = True
+        else:
+            rank_zero_warn(
+                "Setting prepare_data_per_node with the trainer flag is deprecated and will be removed in v1.7.0!"
+                "Please use the property in the DataHooks for setting prepare_data_per_node"
```

Comment:
I wonder if end users are supposed to be aware of `DataHooks`. Should we say `LightningDataModule` here?

Reply:
Good suggestion. I will update this warning. In addition to this question, I'm wondering if we want user to pre-define prepare_data_per_node even when LightningDataModule is not set within trainer?